fuse_models/src/unicycle_2d.cpp

```diff
@@ -228,23 +229,41 @@ void Unicycle2D::onStart()
   state_history_.clear();
 }
 
+void Unicycle2D::generateMotionModelGenerator(
+  const ros::Time& beginning_stamp,
+  const ros::Time& ending_stamp,
+  std::vector<fuse_core::Constraint::SharedPtr>& constraints,
+  std::vector<fuse_core::Variable::SharedPtr>& variables)
+{
+  generateMotionModel(beginning_stamp, ending_stamp, state_history_, device_id_, name(),
+                      process_noise_covariance_, scale_process_noise_, velocity_norm_min_, disable_checks_,
+                      constraints, variables);
+}
+
 void Unicycle2D::generateMotionModel(
   const ros::Time& beginning_stamp,
   const ros::Time& ending_stamp,
+  StateHistory& state_history,
+  const fuse_core::UUID& device_id,
+  const std::string name,
+  fuse_core::Matrix8d process_noise_covariance,
```

Comment:
Do you really need to copy here? Why don't you do the following?:
```suggestion
  const std::string& name,
  const fuse_core::Matrix8d& process_noise_covariance,
```

It looks like you're relying on making a copy of the process noise covariance, so the input doesn't change. I believe it's cleaner to just create a local copy below because there are cases where we return even before using the process nosie covariance, so there's no need to copy always it.